### PR TITLE
Removed support for the EOL galactic distro

### PR DESCRIPTION
**Pull request by maspe36**
The galactic distro was EOL'd on December 9th, 2022. Also worth noting that foxy will be EOL'd in May this year.

https://docs.ros.org/en/humble/Releases.html

### CODE REVIEW

**Comment by maspe36** [comment]
Bumped the version of `setup-ros` and copied a few other changes you had in that PR to see if I could get the builds passing, however now rolling is failing with a `rustc` error that I can't reproduce locally. Any ideas?

**Comment by maspe36** [comment]
@esteve Waiting on approval to merge this, rebased on #302

File: .github/workflows/rust.yml
Comment (esteve) [inline .github/workflows/rust.yml:50]: Can you remove this? I'll prepare a PR that will enable `use-ros2-testing` and will fix the build issue with `rolling`, but this change is not necessary for removing EOL'd distros.

**Review by esteve — CHANGES_REQUESTED** [review]

**Comment by esteve** [comment]
@maspe36 only one comment, but the PR overall looks good. Could you rebase on top of `main`? #303 has been merged. Thanks for all the work :slightly_smiling_face:

File: .github/workflows/rust.yml
Comment (maspe36) [reply .github/workflows/rust.yml]: Done

**Review by esteve — APPROVED** [review]
@maspe36 Thanks!

**Pull request merged**
